types/device.go

```diff
@@ -92,6 +93,13 @@ func (d *DeviceRequest) DecodeMapstructure(value interface{}) error {
 		d.IDs = ids
 		d.Count = DeviceCount(len(ids))
 	}
+
+	if options, ok := v["options"].(map[string]any); ok {
+		d.Options = Mapping{}
```

Comment:
Can't we do this even if there isn't any option defined to avoid nil issue on the client side?